src/modules/network.cpp

```diff
@@ -13,7 +13,14 @@ namespace modules {
   network_module::network_module(const bar_settings& bar, string name_)
       : timer_module<network_module>(bar, move(name_)) {
     // Load configuration values
+    // m_interface could be a regex as well
     m_interface = m_conf.get(name(), "interface", m_interface);
+    string regexpr = m_interface;
+
+    m_interface = net::find_interface_regex(m_interface);
+
+    if (!m_interface.empty())
+      m_log.notice("%s: Matched regex %s with interface %s", name(), regexpr, m_interface);
 
     if (m_interface.empty()) {
```

Comment:
If the regex doesn't match any interface, the module will shut down with the message `Missing parameter "module/wireless.interface-type"` because `m_interface` is empty again here.

If we tried to match using regex, we shouldn't try to use `interface-type`, so something like this:

```cpp
if (!m_interface.empty()) {
  // match regex
} else {
  // find interface type
}
```